tests/unit/WorkflowUtilsTest.ts

```diff
@@ -0,0 +1,529 @@
+                },
+            };
+
+            expect(WorkflowUtils.getApprovalWorkflowApprovers({employees, firstEmail: '1@example.com', personalDetailsByEmail})).toEqual([
+                buildApprover(1, {forwardsTo: '2@example.com'}),
+                buildApprover(2, {forwardsTo: '3@example.com'}),
+                buildApprover(3, {forwardsTo: '4@example.com'}),
+                buildApprover(4),
+            ]);
+            expect(WorkflowUtils.getApprovalWorkflowApprovers({employees, firstEmail: '2@example.com', personalDetailsByEmail})).toEqual([
+                buildApprover(2, {forwardsTo: '3@example.com'}),
+                buildApprover(3, {forwardsTo: '4@example.com'}),
+                buildApprover(4),
+            ]);
+            expect(WorkflowUtils.getApprovalWorkflowApprovers({employees, firstEmail: '3@example.com', personalDetailsByEmail})).toEqual([
+                buildApprover(3, {forwardsTo: '4@example.com'}),
+                buildApprover(4),
+            ]);
+        });
+
+        it('Should return a list of approver when there are forwardsTo', () => {
+            const employees: PolicyEmployeeList = {
+                '1@example.com': {
+                    email: '1@example.com',
+                    forwardsTo: '2@example.com',
+                },
+                '2@example.com': {
+                    email: '2@example.com',
+                    forwardsTo: '3@example.com',
+                },
+                '3@example.com': {
+                    email: '3@example.com',
+                    forwardsTo: '4@example.com',
+                },
+                '4@example.com': {
+                    email: '4@example.com',
+                    forwardsTo: undefined,
+                },
+                '5@example.com': {
+                    email: '5@example.com',
+                    forwardsTo: undefined,
+                },
+            };
+
+            expect(WorkflowUtils.getApprovalWorkflowApprovers({employees, firstEmail: '1@example.com', personalDetailsByEmail})).toEqual([
+                buildApprover(1, {forwardsTo: '2@example.com'}),
+                buildApprover(2, {forwardsTo: '3@example.com'}),
+                buildApprover(3, {forwardsTo: '4@example.com'}),
+                buildApprover(4),
+            ]);
+            expect(WorkflowUtils.getApprovalWorkflowApprovers({employees, firstEmail: '2@example.com', personalDetailsByEmail})).toEqual([
+                buildApprover(2, {forwardsTo: '3@example.com'}),
+                buildApprover(3, {forwardsTo: '4@example.com'}),
+                buildApprover(4),
+            ]);
+            expect(WorkflowUtils.getApprovalWorkflowApprovers({employees, firstEmail: '3@example.com', personalDetailsByEmail})).toEqual([
+                buildApprover(3, {forwardsTo: '4@example.com'}),
+                buildApprover(4),
+            ]);
+        });
+
+        it('Should return a list of approver with circular references', () => {
```

Comment:
In here and few other places where plural `approvers` would make sense you use singular `approver`. Not sure why, but probably a typo?